### ISSUE TITLE
[Player Requests] Licenses

### ISSUE THREAD

**Issue by jpcsupplies** [jpcsupplies/Economy_mod#113]
Spacemarine: Idea for eco-mod: licenses. Ability to build stuff like ships of a certain size etc. good as a motivation for players to make some money. Can be extended to certain blocks or classes of blocks eg. military stuff... Just some random shit I made up. Thought I'd share it with you before forgetting it.
--

Not entirely sure what he means.
Either a licence to permit the construction of a specific class and size ship (presumption being they are restricted or penalised in this regard without a license)
Or a license that allows them to build prefabs for a fixed cost?
Or a contract to build a ship and be paid for the labor?

**Comment by Spcemarine** [comment]
Yeah, basically the first option you mentioned. Eg. when you start on a server you are able to build a ship of Mk I. Mk I is restricted to 50 Blocks for small ships, 0 blocks for large ships and 100 blocks for stations. Also you are only permitted to own one of all grid sizes. Mk II could cost XXX.XXX coins and you are allowed to build (or control) small ships up to 100 blocks, large ships up to 100 blocks and stations up to 300 blocks.

On top of that you can unlock certain classes of blocks on Mk II, eg. the rocket launcher or military blocks in general. Or like the jump drive is unlocked at Mk X. Something else would be that you are allowed to have a certain amount of these blocks. 

With the blocks it's more like paying for patents and with the grids it's more like buying a license for that size. Something like that could be a nice motivation for players to make money in order to reach a higher tier that allows them to make even more money. On a side note: If this is ever going to be implemented, make sure that it is configurable and works with other mods so that admins can set a certain price to enhanced modded blocks (like the titan engine).

I hope this makes clear what I intended to say.

**Comment by jpcsupplies** [comment]
Yer i get a minecraft vibe from this - would have been good with the movecraft or the later air pirates curruption of movecraft plugin on bukkit.

Not sure if you are following all our discussions, but we have a way to value entire ships now; player will soon hopefully be able to use the plugin to handle transfer of ownership buying/selling ships and stations, and be able to 'register" as a trader.  Players are free to handle the ownership side themself now with the /pay command to conclude the sale too.  They can use the plugin to handle component trades with each other too.

**Comment by Spcemarine** [comment]
To be honest, I'm not really following your discussions. As a former admin of a semi-public server I just have the feeling that if I installed the mod I would have to give the players a little more motivation than "just" enhanced trade. If they can't buy anything else with the ingame currency most players won't use any of the mod's features. They'd just make money for fun when they already are in a endgame like situation. So I'd like to give them something else than "just" resource trade (even components or ships are crafted out of resources, of course they're not the same but with the necessary ingots you cans build or.craft them quite easiely). I'm quite sure that if everyone had a reason to have a small amount of money on him it would boost the trade among the players. Additionally it would be really nice to have those features as they would make the process of advancing in the game a bit slower.

**Comment by jpcsupplies** [comment]
Hmmm  actually...  player progression might make a good mission chain for the mission system hud we are playing with for two releases in the future.

So you can build a ship of a certain size but to upgrade size you pay a fee
This unlocks the next size upgrade of being able to make another (or upgrade) ship of a bigger size.  To go up to the next size level again they pay a fee again etc..

Basically what you are saying bit fits in our mission system.. hmm

**Comment by Spcemarine** [comment]
If it fits in, that's perfect then. However, it is hard to implement it right now as there are serveral issues. E.g. how to deal with merges, connectors and landing gears. Also how do you enforce the rules? By preventing them from exceeding the limits with a piece of code? Or a bounty?

I personally think a bounty won't be efficient as it can be very hard to find a player and on top of that I'm not afraid of dying since I can respawn freely... So really enforcing it by code might be the best way logicwise but it's difficult to implement...

**Comment by jpcsupplies** [comment]
Well i dont want economy to become an admin utility, that is what midspace admin mod is for.. players should be free to play how they like - but in the spirit of a role play mod (look at the game transport tycoon which myself and midspace have in common having played)  i would personally aim for something smaller scale.   If we implemented a ship registry (in effect the roleplay aspect of what you propose)  then ships outside of their licenced size would become illegal, or rogue ships.      This means for example we could restrict them accessing parts of the trade network; or if we implement delivery "subsidies"   we could back this up with a reputation system - which means rogue players get "negative" subsidies; making all trades more expensive, and limiting them from trading certain item types.  Downside (or upside?) is this forces rogue players to mine or raid more for supplies.

Or we could adopt a retro Elite and x-game series model, where they are only allowed to trade with pirate stations.   This means allowing players to flag their station as an outlaw friendly station;  - or create trade zones on exploration pirate stations (not so hot on that idea)

Which always runs the risk of "its a trap" if all the guns go hot as soon as he gets too close.
(player sets trade radius to 500.. forcing traders to enter gun range.. flags his station outlaw friendly..  parks a few sensor satelites around...  you get the idea)

As for bounties...  if i went that way  - news system - window pops up with the news feed item: "alert outlaw player/ship X has been sighted near x y z.  All traders are warned to avoid the area." 

If we get the vanity names system going it could plug into that and add a navigation hazard GPS point to everyones database. Which of course means any bored player with a warship will immediately light their jump drives.. and hit that gps.

I am not entirely heartless.. i could only make the warning pop up if a outlaw player ventures too close to any registered trade region.   But make this check quite brutial (say 100km radius lol)

**Comment by jpcsupplies** [comment]
Of course anything this ambitious would be a long way down our todo list if we ever do it.
economy mod is already close to crossing the line from enhancement to total conversion :)
